chart/templates/deployment.yaml

```diff
@@ -25,9 +25,7 @@ spec:
 {{- end }}
         prometheus.io/scrape: "true"
     spec:
-{{- if .Values.serviceAccount.create }}
-      serviceAccount: {{ .Values.app.name }}
-{{- end }}
+      serviceAccountName: {{.Values.serviceAccount.name | default .Values.app.name | quote }}
```

Comment:
This works for both your use case (manually-created SA) and what I'm treating as the default (chart-owned SA), but it won't work if you don't need an SA at all (e.g. your nodes already have SQS permissions, bad idea though that may be).

How about this instead:
```yaml
{{- if or .Values.serviceAccount.name .Values.serviceAccount.create }}
      serviceAccountName: {{ .Values.serviceAccount.name | default .Values.app.name | quote }}
{{- end }}
```

That should take care of all three use cases.